core/src/main/java/org/elasticsearch/action/support/ActiveShardCount.java

```diff
@@ -140,17 +137,16 @@ public static ActiveShardCount parseString(final String str) {
      * Returns true iff the given cluster state's routing table contains enough active
      * shards to meet the required shard count represented by this instance.
      */
-    public boolean enoughShardsActive(final ClusterState clusterState, final String indexName, final Settings settings) {
-        Objects.requireNonNull(clusterState);
-        Objects.requireNonNull(indexName);
-        Objects.requireNonNull(settings);
+    public boolean enoughShardsActive(final ClusterState clusterState, final String indexName) {
         if (this == ActiveShardCount.NONE) {
             // not waiting for any active shards
             return true;
         }
+        final IndexMetaData indexMetaData = clusterState.metaData().index(indexName);
+        assert indexMetaData != null;
```

Comment:
Its a bit of a weird situation, as it would mean the cluster state was independently monitored to know the index exists, and then a delete index request was issued while still waiting for shards after the create request.  We should definitely still account for it, but should the create index response indicate that this occurred, instead of just returning `{ acknowledged: true, timed_out_waiting_for_shards: false }`?